### ISSUE TITLE
SwiftFormat breaks optional generic return value

### ISSUE THREAD

**Issue by lhunath** [nicklockwood/SwiftFormat#1441]
Given the following code:
```
func foo() -> Optional<some Publisher<String, Never>> {
    fatalError()
}
```
SwiftFormat will rewrite as:
```
func foo() -> some Publisher<String, Never>? {
    fatalError()
}
```
This results in illegal Swift code:
```
Use of protocol 'Publisher' as a type must be written 'any Publisher'
```
(The error message is misleading, the correct code was the original; do not change `some` to `any`)

```
$ swiftformat --version
0.51.2
$ swift --version
swift-driver version: 1.62.15 Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
```

**Comment by nicklockwood** [comment]
I think the issue is that it needs to be written as `(some Publisher<String, Never>)?`.

Obviously swiftformat shouldn't introduce a breaking change here, but until the bug is fixed, manually adding the parens might be a workaround?

**Comment by nicklockwood** [comment]
@lhunath fixed in 0.51.9
